Core/ContentBlockerConfigurationUserDefaults.swift

```diff
@@ -29,8 +29,19 @@ public class ContentBlockerProtectionUserDefaults: ContentBlockerProtectionStore
 
     private let suiteName: String
 
+    // This variable should be confined to the Main Thread
+    private var tempUnprotectedDomains: [String]?
```

Comment:
I think we should consider changing this to a computed property which uses the app dependency provider, and modify the storage cache to have a cached version of this list which it reads lazily from file storage.

We also inject the storage cache in to this class and load the temp list there so could do the same there too.

  https://github.com/duckduckgo/iOS/blob/develop/Core/ContentBlockerRulesUserScript.swift#L51

Reply:
@samsymons pointed out this class is in Core.

I think we should consider flattening core into the app and remodularising to put the right access levels on these config classes.

Extensions that use core classes can just reference those files directly via their target. 

I’ll have a think about that.

Reply:
I agree with all of your comments - this was a quick fix to a performance problem. When we started diving into that we figured out there is more to that, which I pointed in in the original task and even created a follow up one: https://app.asana.com/0/414709148257752/1200571576333117 Basically we need to change how we inject these dependencies and how they relate to each other.